test/parallel/test-buffer-indexof.js

```diff
@@ -401,13 +401,38 @@ assert.equal(b.lastIndexOf(0x61, Infinity), 0);
 assert.equal(b.lastIndexOf(0x0), -1);
 
 // Test weird offset arguments.
-// Behaviour should match String.lastIndexOf:
-assert.equal(b.lastIndexOf('b', 0), -1);
-assert.equal(b.lastIndexOf('b', undefined), 1);
-assert.equal(b.lastIndexOf('b', null), -1);
-assert.equal(b.lastIndexOf('b', {}), 1);
-assert.equal(b.lastIndexOf('b', []), -1);
-assert.equal(b.lastIndexOf('b', [2]), 1);
+// The following offsets coerce to NaN, searching the whole Buffer (or String)
+assert.strictEqual(b.lastIndexOf('b', undefined), 1);
+assert.strictEqual(b.lastIndexOf('b', {}), 1);
+
+// The following offsets coerce to 0
+assert.strictEqual(b.lastIndexOf('b', 0), -1);
+assert.strictEqual(b.lastIndexOf('b', null), -1);
+assert.strictEqual(b.lastIndexOf('b', []), -1);
+
+// The following offset coerces to 2, in other words +[2] === 2
+assert.strictEqual(b.lastIndexOf('b', [2]), 1);
+
+// Behavior should match String.lastIndexOf()
```

Comment:
Sure, but `indexOf` is less interesting. Args coercing to 0 and NaN both have the same behavior, searching the whole buffer. Only `lastIndexOf` has this weirdness where, for example, passing an offset of `null` vs `undefined` does different things.